statistics/sample.go

```diff
@@ -29,9 +30,58 @@ import (
 	"github.com/pingcap/tipb/go-tipb"
 )
 
+// SampleItem is an item of sampled column value.
+type SampleItem struct {
+	// Value is the sampled column value.
+	Value types.Datum
+	// Ordinal is original position of this item in SampleCollector before sorting. This
+	// is used for computing correlation.
+	Ordinal int
+}
+
+// SortSampleItems sorts a slice of SampleItem.
+func SortSampleItems(sc *stmtctx.StatementContext, items []*SampleItem) error {
+	sorter := sampleItemSorter{items: items, sc: sc}
+	sort.Sort(&sorter)
+	return sorter.err
+}
+
+// extractSampleItemsDatums is for test purpose only to extract Datum slice
+// from SampleItem slice.
+func extractSampleItemsDatums(items []*SampleItem) []types.Datum {
```

Comment:
It's better to put this function in `fmsketch_test.go`?

Reply:
Updated, PTAL